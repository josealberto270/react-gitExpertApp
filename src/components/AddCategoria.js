import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategoria = ({ setCategorias }) => {
  const [inputValue, setInputValue] = useState('')

  const handleSubmit = (e) => {
    e.preventDefault()
    const _inputValue = inputValue.trim()
    if (_inputValue.length > 2) {
      setCategorias(cats => [_inputValue, ...cats])
      setInputValue('')
    }
  }

  const handleInputChange = (e) => {
    setInputValue(e.target.value)
  }

  return (
    <form onSubmit={handleSubmit}>
      <p>{inputValue}</p>
      <input
        type="text"
        value={inputValue}
        onChange={handleInputChange}
      />
    </form>
  )
}

AddCategoria.propTypes = {
  setCategorias: PropTypes.func.isRequired
}