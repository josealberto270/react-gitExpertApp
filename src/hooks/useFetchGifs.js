import { useEffect, useState } from "react"
import { getGif } from '../helpers/getGifs'

export const useFetchGifs = (categoria) => {
  const [state, setState] = useState({
    data: [],
    loading: true,
  })

  useEffect(() => {
    getGif(categoria).then(imgs =>
      setTimeout(() => {
        setState({
          data: imgs,
          loading: false
        })

      }, 3000)
    )
  }, [categoria])

  return state //data:[],loading:true
}
