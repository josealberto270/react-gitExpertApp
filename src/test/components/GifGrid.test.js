import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import { GifGrid } from '../../components/GifGrid'
import { useFetchGifs } from '../../hooks/useFetchGifs'
jest.mock('../../hooks/useFetchGifs')


describe('Pruebas en el componente GifGrid', () => {


  test('debe mostrar <GifGrid /> correctamente', () => {

    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    })

    const wrapper = shallow(<GifGrid categoria={'Naruto'} />)
    expect(wrapper).toMatchSnapshot()
  })

  test('debe de mostrar items cuando se cargan imagenes useFetchgifs ', () => {

    const gifs = [{
      id: '1234',
      url: 'http://localhost/cualquiercosa/img.png',
      title: 'cualquier cosa'
    }]

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    })

    const wrapper = shallow(<GifGrid categoria={'Naruto'} />)
    expect(wrapper).toMatchSnapshot()
  })


})
