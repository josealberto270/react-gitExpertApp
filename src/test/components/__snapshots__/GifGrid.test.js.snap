// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifGrid debe de mostrar items cuando se cargan imagenes useFetchgifs  1`] = `
<Fragment>
  <h3
    className=" animate__animated animate__fadeIn"
  >
    Naruto
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="1234"
      key="1234"
      title="cualquier cosa"
      url="http://localhost/cualquiercosa/img.png"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en el componente GifGrid debe mostrar <GifGrid /> correctamente 1`] = `
<Fragment>
  <h3
    className=" animate__animated animate__fadeIn"
  >
    Naruto
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
